File: goty-server/src/main/java/com/aleinin/goty/game/GameCountService.java
package com.aleinin.goty.game;

import lombok.RequiredArgsConstructor;
import org.elasticsearch.client.RequestOptions;
import org.elasticsearch.client.RestHighLevelClient;
import org.elasticsearch.client.core.CountRequest;
import org.elasticsearch.index.query.QueryBuilders;
import org.springframework.stereotype.Component;

import java.io.IOException;

@Component
@RequiredArgsConstructor
class GameCountService {

    private final RestHighLevelClient elasticsearchClient;

    long count() throws IOException {
        var countRequest = new CountRequest("games")
                .query(QueryBuilders.matchAllQuery());
        return elasticsearchClient
                .count(countRequest, RequestOptions.DEFAULT)
                .getCount();
    }
}

File: goty-server/src/main/kotlin/com/aleinin/goty/game/GameCountService.kt
package com.aleinin.goty.game

import org.elasticsearch.client.RequestOptions
import org.elasticsearch.client.RestHighLevelClient
import org.elasticsearch.client.core.CountRequest
import org.elasticsearch.index.query.QueryBuilders
import org.springframework.stereotype.Component

@Component
class GameCountService(private val elasticsearchClient: RestHighLevelClient) {

    fun count() =
        elasticsearchClient.count(
            CountRequest("games").query(QueryBuilders.matchAllQuery()),
            RequestOptions.DEFAULT
        ).count
}
